1.3 announcement blog (#4929)
istio/istio.io: content/en/blog/2019/announcing-1.3/index.md

{{< relnote linktonote="true" >}}

The theme of 1.3 is User Experience - improving the experience of new users adopting Istio, debugging problems and supporting more applications out of the box. 

Users are adopting Istio faster than ever before and we’ve been listening to the feedback! We’ve heard that Istio can be complex and challenging to adopt and we've been working hard to make improvements. 

We started by forming a new User Experience team earlier this year to lead improvements across all components of Istio. Everyone is invited to join the working group meetings and share your thoughts. We’ve begun delivering many notable improvements, some of which are ready for you to use in 1.3 and some in experimental state. 

In the past, we required you to explicitly declare the protocol for service ports. This was a common gotcha as users forgot to name their ports when adding their application to the mesh. By default, Pilot now has protocol sniffing which removes this requirement! Along the same lines, another obsolete requirement is the need to define all the pod ports (`containerPort`) for each container. Istio will now capture all inbound ports by default.

> If you haven't tried out `istioctl` yet, now is a great time to start! Formerly experimental sub-commands `metrics`, `dashboard` and `convert-ingress` have been promoted and are now supported Istio features. In addition, several new istioctl experimental subcommands have been added to help improve the usability of the system.

Reviewer: In addition, these commands should link to the relevant doc pages. Just find the right link in /docs/reference/commands/istioctl/

The Istio team is also constantly working on improving the control plane performance and reducing its footprint. In this release, Pilot performance has been significantly improved resulting in as much as a 90% CPU usage savings!

Locality aware Load Balancing is another feature which graduated from experimental to default in this release. Istio can now take advantage of existing locality information to prioritize load balancing pools and favor sending requests to the closest backends.

> Control plane monitoring has been greatly enhanced with additional metrics to monitor configuration state, Sidecar Injector webhook, a new Grafana dashboard for Citadel, and improvements to the Pilot dashboard.

Reviewer 2: 2nd point should read as Added metrics for sidecar injector

> 

Reviewer 3: Could we add a paragraph and a list highlighting some of the work of the docs team to share the biggest pieces we crafted for this release? @geeknoid @sdake @frankbu WDYT? I've added the two biggest items I could think of in my suggestion, would you include other highlights?

```suggestion
We've heard how important it is for you to have accurate and up-to-date information in our documentation. The documentation team continuously works to provide with the information you need to be successful. In Istio 1.3, we made many improvements but the following are some highlights:

- Added the new [Istio Deployment Models concept](/docs/concepts/deployment-models/) to help you decide what deployment model suits your needs.
- Organized the content in of our [Operations Guide](/docs/ops/) and created a [section with all troubleshooting tasks](/docs/ops/troubleshooting) to help you find the information you seek faster. 

Thanks to our community testers, our [documentation](/docs/) is tested for every release including Istio 1.3!
```

---
title: Announcing Istio 1.3
subtitle: Major Update
description: Istio 1.3 release announcement.
publishdate: 2019-09-10
attribution: The Istio Team
release: 1.3.0
---

We are pleased to announce the release of Istio 1.3!  

{{< relnote linktonote="true" >}}

The theme of Istio 1.3 is User Experience:

- Improve the experience of new users adopting Istio
- Improve the experience of users debugging problems
- Support more applications without any additional configuration

Every few releases, the Istio team gets excited as we make dramatic improvements to usability, APIs, and the overall system performance. This release is one of those where the team is excited to roll out some key updates:

*Intelligent protocol detection*: In past releases, you had to explicitly declare the protocol for service ports by using a special port naming format. This requirement caused problems for users that didn't name their ports when they added their application to the mesh. Starting with 1.3, the protocol on the wire will be automatically detected (as HTTP or TCP), eliminating the need to name ports explicitly. We will be adding support for additional [non-HTTP] protocols in future releases. 

> *Container ports are not required*: In the past, Istio required that pods explicitly declare the Kubernetes `containerPort` for each container as a security measure against trampolining traffic. Istio 1.3 has secure and simpler way of handling all inbound traffic on any port into a Pod without requiring `containerPort` declarations at all. Along the way, we have completely eliminated infinite loops caused in the IPtables rules when pods send traffic to themselves. 

Reviewer 4: ```suggestion
*Container ports are not required*: In the past, Istio required that pods explicitly declare the Kubernetes `containerPort` for each container as a security measure against trampolining traffic. Istio 1.3 has a secure and simpler way of handling all inbound traffic on any port into a Pod without requiring any `containerPort` declarations at all. Along the way, we have completely eliminated infinite loops caused in the IPtables rules when pods send traffic to themselves. 
```

> *Enhanced EnvoyFilters*: While Istio 1.3 focuses on out of box usability, it also caters to the expert users who would like to use advanced features in Envoy that Istio does note expose via its Networking APIs. We have enhanced the `EnvoyFilter` API to allow users to fully customize the Envoy HTTP route configuration returned by RDS, HTTP/TCP listeners and their filter chains returned by LDS, and the set of clusters returned by CDS. 

Reviewer 4: ```suggestion
*Enhanced EnvoyFilters*: While Istio 1.3 focuses on out of box usability, it also caters to the expert users who would like to use advanced features in Envoy that Istio does not expose via its Networking APIs. We have enhanced the `EnvoyFilter` API to allow users to fully customize the Envoy HTTP route configuration returned by RDS, HTTP/TCP listeners and their filter chains returned by LDS, and the set of clusters returned by CDS.
```

---
title: Announcing Istio 1.3
subtitle: Major Update
description: Istio 1.3 release announcement.
publishdate: 2019-09-10
attribution: The Istio Team
release: 1.3.0
---

We are pleased to announce the release of Istio 1.3!  

{{< relnote linktonote="true" >}}

The theme of Istio 1.3 is User Experience:

- Improve the experience of new users adopting Istio
- Improve the experience of users debugging problems
- Support more applications without any additional configuration

Every few releases, the Istio team gets excited as we make dramatic improvements to usability, APIs, and the overall system performance. This release is one of those where the team is excited to roll out some key updates:

*Intelligent protocol detection*: In past releases, you had to explicitly declare the protocol for service ports by using a special port naming format. This requirement caused problems for users that did not name their ports when they added their applications to the mesh. Starting with 1.3, the protocol on the wire will be automatically detected (as HTTP or TCP), eliminating the need to name ports explicitly. We will be adding support for additional [non-HTTP] protocols in future releases. 

> *Mixer-less telemetry*: Yes you read that right! We had already turned off the `istio-policy` service by default as we implemented most of the common security policies such as RBAC directly in Envoy. We are now on track to migrate most of the mixer's telemetry functionality as well into Envoy. As of this release, if all you care about is telemetry for HTTP services, we have enhanced the Istio proxy to emit HTTP metrics directly to Prometheus, without requiring the `istio-telemetry` service to enrich the information. You can experiment with the mixer-less telemetry feature for HTTP services by following these [instructions](https://github.com/istio/istio/wiki/Mixerless-HTTP-Telemetry). We will be polishing this feature in the coming months including adding telemetry support for TCP services when Istio mTLS is being used. 

Reviewer 5: Make it clear that it is *experimental* in 1.3. 
`*Mixer-less http telemetry (experimental)*`

*Mixer-less telemetry (experimental)*: Yes you read that right! We had already turned off the `istio-policy` service by default as we implemented most of the common security policies such as RBAC directly in Envoy. We are now on track to migrate most of the mixer's telemetry functionality as well into Envoy. As of this release, if all you care about is telemetry for HTTP services, we have enhanced the Istio proxy to emit HTTP metrics directly to Prometheus, without requiring the `istio-telemetry` service to enrich the information. You can experiment with the mixer-less telemetry feature for HTTP services by following these [instructions](https://github.com/istio/istio/wiki/Mixerless-HTTP-Telemetry). We will be polishing this feature in the coming months including adding telemetry support for TCP services when Istio mTLS is being used. 

> *Container ports are no longer required*: In the past, Istio required that pods explicitly declare the Kubernetes `containerPort` for each container as a security measure against trampolining traffic. Istio 1.3 has a secure and simpler way of handling all inbound traffic on any port into a Pod without requiring the `containerPort` declarations at all. We have also completely eliminated infinite loops caused in the IPtables rules when pods send traffic to themselves. 

Reviewer 4: ```suggestion
*Container ports are no longer required*: In the past, Istio required that pods explicitly declare the Kubernetes `containerPort` for each container as a security measure against trampolining traffic. Istio 1.3 has a secure and simpler way of handling all inbound traffic on any port into a Pod without requiring the `containerPort` declarations. We have also completely eliminated infinite loops caused in the IPtables rules when pods send traffic to themselves. 
```

*Customize generated Envoy configuration*: While Istio 1.3 focuses on out of box usability, it also caters to the expert users who would like to use advanced features in Envoy that Istio does not expose via its Networking APIs. We have enhanced the `EnvoyFilter` API to allow users to fully customize the Envoy HTTP route configuration returned by RDS, HTTP/TCP listeners and their filter chains returned by LDS, and the set of clusters returned by CDS. You get the best of both worlds - leverage  Istio to integrate with Kubernetes, handle large fleets of Envoys in an efficient manner and still retain the ability to customize the generated Envoy configuration to meet specific requirements within your infrastructure.

*Other enhancements*:

> - `istioctl` has gained a plethora of debugging features that will highlight various issues in your mesh installation. Checkout the `istioctl` [reference manual](/docs/reference/commands/istioctl/) for the set of all supported features.

Reviewer 4: ```suggestion

- `istioctl` has gained a plethora of debugging features that will highlight various issues in your mesh installation. Checkout the `istioctl` [reference manual](/docs/reference/commands/istioctl/) for the set of all supported features.
```

---
title: Announcing Istio 1.3
subtitle: Major Update
description: Istio 1.3 release announcement.
publishdate: 2019-09-10
attribution: The Istio Team
release: 1.3.0
---

We are pleased to announce the release of Istio 1.3!  

{{< relnote linktonote="true" >}}

The theme of Istio 1.3 is User Experience:

- Improve the experience of new users adopting Istio
- Improve the experience of users debugging problems
- Support more applications without any additional configuration

> Every few releases, the Istio team gets excited as we make dramatic improvements to usability, APIs, and the overall system performance. This release is one of those where the team is excited to roll out some key updates:

Reviewer: Remove the : and end with a .

> *Intelligent protocol detection*: In past releases, you had to explicitly declare the protocol for service ports by using a special port naming format. This requirement caused problems for users that did not name their ports when they added their applications to the mesh. Starting with 1.3, the protocol on the wire will be automatically detected (as HTTP or TCP), eliminating the need to name ports explicitly. We will be adding support for additional [non-HTTP] protocols in future releases. 

Reviewer: Replace all the paragraph lead-ins with real sections:

## Intellifent protos detection

Reviewer 6: This is not entirely accurate as I just hit this today.  User will still need to name the ports when there are more than 1 port.  It is just we don't enforce more requirements than what K8s has and there is no need to prefix the name with protocol.   Also, this is only for http and tcp for 1.3.  So it should be something like "eliminating the need to prefix the port name explicitly for HTTP or TCP."

Author: Done

> *Mixer-less telemetry (experimental)*: Yes you read that right! We had already turned off the `istio-policy` service by default as we implemented most of the common security policies such as RBAC directly in Envoy. We are now on track to migrate most of the mixer's telemetry functionality as well into Envoy. As of this release, if all you care about is telemetry for HTTP services, we have enhanced the Istio proxy to emit HTTP metrics directly to Prometheus, without requiring the `istio-telemetry` service to enrich the information. You can experiment with the mixer-less telemetry feature for HTTP services by following these [instructions](https://github.com/istio/istio/wiki/Mixerless-HTTP-Telemetry). We will be polishing this feature in the coming months including adding telemetry support for TCP services when Istio mTLS is being used. 

Reviewer 6: Is it appropriate to highlight this and point to a wiki page without this featured in istio.io?  I'm a little concerned with this especially I can't get telemetry working right now for RC 3.

Reviewer 5: which part of telemetry does not work in 1.3-rc3 ? is there an issue?

Author: Resolving this for now. Please start another comment if we decide to remove this from the blog based on the outcome of the issue investigation.

> *Container ports are no longer required*: In the past, Istio required that pods explicitly declare the Kubernetes `containerPort` for each container as a security measure against trampolining traffic. Istio 1.3 has a secure and simpler way of handling all inbound traffic on any port into a Pod without requiring the `containerPort` declarations. We have also completely eliminated infinite loops caused in the IPtables rules when pods send traffic to themselves.

Reviewer 6: We need to be careful about stating this.  This can be required from k8s.   It is no longer required for istio.

Reviewer 7: yes thats why I framed it as "Istio required that.." and then, "Istio no longer requires"..

*Other enhancements*:

- `istioctl` has gained a plethora of debugging features that will highlight various issues in your mesh installation. Checkout the `istioctl` [reference manual](/docs/reference/commands/istioctl/) for the set of all supported features.

- Locality aware load balancing graduated from experimental to default in this release too. Istio can now take advantage of existing locality information to prioritize load balancing pools and favor sending requests to the closest backends.

- Better support for headless services with Istio mTLS 

- We enhanced control plane monitoring in the following ways:
  - Added new metrics to monitor configuration state
  - Added metrics for sidecar injector
  - Added a new Grafana dashboard for Citadel
  - Improved the Pilot dashboard to expose additional key metrics

- Added the new [Istio Deployment Models concept](/docs/concepts/deployment-models/) to help you decide what deployment model suits your needs.

- Organized the content in of our [Operations Guide](/docs/ops/) and created a [section with all troubleshooting tasks](/docs/ops/troubleshooting) to help you find the information you seek faster.

> Thanks to our community testers, our [documentation](/docs/) is tested for every release including Istio 1.3!

Reviewer: I'd remove this paragraph. I certainly hope stuff is tested before release, calling it out seems redundant.

See the [release notes](/about/notes/1.3) for the complete list of changes.

As always, there is a lot happening in the [Community Meeting](https://github.com/istio/community#community-meeting); join us every other Thursday at 11 AM Pacific.

> Join one of our [Work Groups](https://github.com/istio/community/blob/master/WORKING-GROUPS.md) and help us make Istio better.

Reviewer: "Work Groups" -> working groups

---
title: Announcing Istio 1.3
subtitle: Major Update
description: Istio 1.3 release announcement.
publishdate: 2019-09-10
attribution: The Istio Team
release: 1.3.0
---

We are pleased to announce the release of Istio 1.3!

{{< relnote linktonote="true" >}}

The theme of Istio 1.3 is User Experience:

- Improve the experience of new users adopting Istio
- Improve the experience of users debugging problems
- Support more applications without any additional configuration

> Every few releases, the Istio team gets excited as we make dramatic improvements to usability, APIs, and the overall system performance. This release is one of those where the team is excited to roll out some key updates.

Reviewer 3: ```suggestion
Every few releases, the Istio team delivers dramatic improvements to usability, APIs, and the overall system performance. Istio 1.3 is one such release, and the team is very excited to roll out some key updates.
```

## Intelligent protocol detection

> In past releases, you had to explicitly declare the protocol for service ports by using a special port naming format. This requirement caused problems for users that did not name their ports when they added their applications to the mesh. Starting with 1.3, the protocol on the wire will be automatically detected (as HTTP or TCP), eliminating the need to name ports explicitly. We will be adding support for additional [non-HTTP] protocols in future releases.

Reviewer 3: ```suggestion
In past releases, you had to use a special port naming format to explicitly declare the protocol for service ports. This requirement caused problems for users that did not name their ports when they added their applications to the mesh. Starting with 1.3, the protocol is automatically detected as HTTP or TCP. This detection eliminates the need to name ports explicitly. We are adding support for additional [non-HTTP] protocols in future releases.
```

## Mixer-less telemetry (experimental)

> Yes you read that right! We had already turned off the `istio-policy` service by default as we implemented most of the common security policies such as RBAC directly in Envoy. We are now on track to migrate most of the mixer's telemetry functionality as well into Envoy. As of this release, if all you care about is telemetry for HTTP services, we have enhanced the Istio proxy to emit HTTP metrics directly to Prometheus, without requiring the `istio-telemetry` service to enrich the information. You can experiment with the mixer-less telemetry feature for HTTP services by following these [instructions](https://github.com/istio/istio/wiki/Mixerless-HTTP-Telemetry). We will be polishing this feature in the coming months including adding telemetry support for TCP services when Istio mutual TLS is being used.

Reviewer 3: ```suggestion
Yes, you read that right! We implemented most of the common security policies directly in Envoy, policies such as RBAC. We previously turned off the `istio-policy` service by default and are now on track to migrate most of Mixer's telemetry functionality into Envoy as well. In this release, we have enhanced the Istio proxy to emit HTTP metrics directly to Prometheus, without requiring the `istio-telemetry` service to enrich the information. This enhancement is great if all you care about is telemetry for HTTP services. Follow the [Mixer-less HTTP telemetry instructions](https://github.com/istio/istio/wiki/Mixerless-HTTP-Telemetry) to experiment with this feature. We are polishing this feature in the coming months to add telemetry support for TCP services when you enable Istio mutual TLS.
```

## Container ports are no longer required

> In the past, Istio required that pods explicitly declare the Kubernetes `containerPort` for each container as a security measure against trampolining traffic. Istio 1.3 has a secure and simpler way of handling all inbound traffic on any port into a Pod without requiring the `containerPort` declarations. We have also completely eliminated infinite loops caused in the IPtables rules when pods send traffic to themselves.

Reviewer 3: ```suggestion
Previous releases required that pods explicitly declare the Kubernetes `containerPort` for each container as a security measure against trampolining traffic. Istio 1.3 has a secure and simpler way of handling all inbound traffic on any port into a {{< gloss >}}workload instance{{< /gloss >}} without requiring the `containerPort` declarations. We have also completely eliminated the infinite loops caused in the IP tables rules when workload instances send traffic to themselves.
```

## Customize generated Envoy configuration

> While Istio 1.3 focuses on out of box usability, it also caters to the expert users who would like to use advanced features in Envoy that Istio does not expose via its Networking APIs. We have enhanced the `EnvoyFilter` API to allow users to fully customize the Envoy HTTP route configuration returned by RDS, HTTP/TCP listeners and their filter chains returned by LDS, and the set of clusters returned by CDS. You get the best of both worlds - leverage  Istio to integrate with Kubernetes, handle large fleets of Envoys in an efficient manner and still retain the ability to customize the generated Envoy configuration to meet specific requirements within your infrastructure.

Reviewer 6: Is this also experimental?  I don't see much doc or testing around this.

Reviewer 7: These are API improvements.  Do we have to prefix everything in the thing with experimental? in that case, we might as well frame that sentence as "the following are the experimental features".. but then saying protocol sniffing is experimental and then stating that it gets enabled by default is not nice.

> While Istio 1.3 focuses on out of box usability, it also caters to the expert users who would like to use advanced features in Envoy that Istio does not expose via its Networking APIs. We have enhanced the `EnvoyFilter` API to allow users to fully customize the Envoy HTTP route configuration returned by RDS, HTTP/TCP listeners and their filter chains returned by LDS, and the set of clusters returned by CDS. You get the best of both worlds - leverage  Istio to integrate with Kubernetes, handle large fleets of Envoys in an efficient manner and still retain the ability to customize the generated Envoy configuration to meet specific requirements within your infrastructure.

Reviewer 3: ```suggestion
While Istio 1.3 focuses on usability, expert users can use advanced features in Envoy that are not part of the Istio Networking APIs. We enhanced the `EnvoyFilter` API to allow users to fully customize:

- The Envoy HTTP route configuration returned by the RDS, HTTP/TCP listeners
- Their filter chains returned by LDS
- The set of clusters returned by CDS. 

You get the best of both worlds:

Leverage Istio to integrate with Kubernetes and handle large fleets of Envoys in an efficient manner, while you still can customize the generated Envoy configuration to meet specific requirements within your infrastructure.
```

## Other enhancements

> - `istioctl` has gained a plethora of debugging features that will highlight various issues in your mesh installation. Checkout the `istioctl` [reference manual](/docs/reference/commands/istioctl/) for the set of all supported features.

Reviewer 3: ```suggestion
- `istioctl` has gained many debugging features to help you highlight various issues in your mesh installation. Checkout the `istioctl` [reference manual](/docs/reference/commands/istioctl/) for the set of all supported features.
```

> - Locality aware load balancing graduated from experimental to default in this release too. Istio can now take advantage of existing locality information to prioritize load balancing pools and favor sending requests to the closest backends.

Reviewer 3: ```suggestion
- Locality aware load balancing graduated from experimental to default in this release too. Istio now takes advantage of existing locality information to prioritize load balancing pools and favor sending requests to the closest backends.
```

publishdate: 2019-09-10
attribution: The Istio Team
release: 1.3.0
---

We are pleased to announce the release of Istio 1.3!

{{< relnote linktonote="true" >}}

The theme of Istio 1.3 is User Experience:

- Improve the experience of new users adopting Istio
- Improve the experience of users debugging problems
- Support more applications without any additional configuration

Every few releases, the Istio team delivers dramatic improvements to usability, APIs, and the overall system performance. Istio 1.3 is one such release, and the team is very excited to roll out some key updates.

## Intelligent protocol detection (experimental)

In past releases, you had to use a special port naming format to explicitly declare the protocol for service ports. This requirement caused problems for users that did not name their ports when they added their applications to the mesh. Starting with 1.3, the protocol for outbound traffic is automatically detected as HTTP or TCP when the ports are not named according to Istio's conventions. We will be polishing this feature in the upcoming releases with support for protocol sniffing on inbound traffic as well as identifying protocols other than HTTP.

> 

Reviewer 6: as much as i love this feature, i would prefer not to mention it as included in the release as it generates more confusion IMHO.  probably good to mention as a forward looking statement.

@rshriram @howardjohn

Reviewer 8: I understand what you mean. I would lean towards keeping it though. Its pretty clear (to me) that it is partially supported and we are working on fixing it

Reviewer 6: If we keep it, let's reword this... we are not eliminating the requirement of special port naming format for this release but are working towards that... this seems to indicate it is eliminated but it is not:

```In past releases, you had to use a special port naming format to explicitly declare the protocol for service ports.```

## Mixer-less telemetry (experimental)

> Yes, you read that right! We implemented most of the common security policies directly in Envoy, policies such as RBAC. We previously turned off the `istio-policy` service by default and are now on track to migrate most of Mixer's telemetry functionality into Envoy as well. In this release, we have enhanced the Istio proxy to emit HTTP metrics directly to Prometheus, without requiring the `istio-telemetry` service to enrich the information. This enhancement is great if all you care about is telemetry for HTTP services. Follow the [Mixer-less HTTP telemetry instructions](https://github.com/istio/istio/wiki/Mixerless-HTTP-Telemetry) to experiment with this feature. We are polishing this feature in the coming months to add telemetry support for TCP services when you enable Istio mutual TLS.

Reviewer 6: I think linking to WIKI (untested) and highlight in release blog is not appropriate.  cc @sdake and @brian-avery for thoughts.
